### DISCUSSION TITLE
how do I have two classes(HID, CDC) alternatively in my application?

### DISCUSSION THREAD

**Discussion by nachiketathakur** [hathach/tinyusb#1428] [Q&A]
I am using RP2040(Pico Development Board).
In my application, I want to do something like this: 
userInput; 
tusb_init(); 
if (userInput == 0) 
initialize_cdc_class(); 
if(userInput == 1)
initialize_hid_class(); 

Is it possible to do this? And is it the tud_task() which initializes the device stack as per descriptors or is it the tusb_init() ? If it is the tusb_init() then how can I make changes so that I can alternatively have two classes in the application?

**Comment by hathach** [comment]
I don’t quite understand what you want to achieve. Do you want to have device dynamically enumerated as different class/type based on user action (or any code logic) ? If that is the case check out `dynamic_configuration` example 
https://github.com/hathach/tinyusb/blob/9c8c5c1c53214b8954bb28c33a496b5b423d5ecc/examples/device/dynamic_configuration/src/usb_descriptors.c#L189 which present itself as msc or midi depending on whether the on-board button is pressed when enumerating (plugging)

Configuration descriptor is what host and device agree to work together. If you want to change that when host ask when enumerating. After enumeration any changes required an detach/re-attach.
